rust/gui-client/src-tauri/src/client/settings.rs

```diff
@@ -4,16 +4,20 @@
 use crate::client::gui::{self, ControllerRequest, Managed};
 use anyhow::{Context, Result};
 use atomicwrites::{AtomicFile, OverwriteBehavior};
+use connlib_shared::messages::ResourceId;
 use firezone_headless_client::known_dirs;
 use serde::{Deserialize, Serialize};
-use std::{io::Write, path::PathBuf, time::Duration};
+use std::{collections::HashSet, io::Write, path::PathBuf, time::Duration};
 use tokio::sync::oneshot;
 use url::Url;
 
 #[derive(Clone, Deserialize, Serialize)]
 pub(crate) struct AdvancedSettings {
     pub auth_base_url: Url,
     pub api_url: Url,
+    // TODO: Will this cause problems when we add the ability to enable / disable Resources on the client side? At least the Internet Resource will need that.
+    #[serde(default)]
+    pub favorite_resources: HashSet<ResourceId>,
```

Comment:
Jamil recommended a different format but I was thinking we could probably get away with SoA / ECS style, so one `HashSet` for favorites and another `HashSet` for disabled, instead of a vector of structs.

It's JSON so it doesn't matter a ton anyway - We're parsing the whole thing from disk every time no matter what.

Reply:
I am in favor of SoA. Just bear in mind that `HashSet`s don't have an order but I guess we need to sort somewhere else in the GUI explicitly anyway.

Reply:
It is fine for now, once we want to change the format we probably want to add a version field or something.